IPython/core/display.py

```diff
@@ -1059,6 +1078,9 @@ def __init__(self, data=None, url=None, filename=None, format=None,
                     format = self._FMT_JPEG
                 if ext == u'png':
                     format = self._FMT_PNG
+                if ext == u'gif':
+                    # use PNG format until we understand why GIF is not working
+                    format = self._FMT_PNG
```

Comment:
I think you're right. I created [a branch](https://github.com/gnestor/notebook/tree/image-gif) for this very purpose but the issue I am running into is that when running the following:

```py
with open('../giphy.gif','rb') as f:
    display(Image(data=f.read(), format="gif"))
```

the data-uri looks like `data:image/gif;base64,GIF89a�w�  2MekdWE71Sv�������gWN` 

When using `format="png"`, the data-uri looks like `data:image/png;base64,R0lGODlh9AF3AfcAADJNZWtkV0U3MVN2n6qai+rWyWd...`. 

For some reason, the PNG format encoding the data different than the GIF format but I can't find any code in ipython or the notebook that does any encoding. 

Any ideas?